### DISCUSSION TITLE
Event for when the menu is opening for the select components

### DISCUSSION THREAD

**Discussion by MoaLeonhardt** [material-components/material-web#4789] [Ideas]
Concerning the select components (md-outlined-select and md-filled-select).

The regular HTML select component lacks events for when the select menu is opened. However, it looks like md-menu has an @opening event. If that event would be available on the material-web components md-outlined-select and md-filled-select it would be very useful.

For example, I'm using @lit-labs/task to create a select component where the menu data is not loaded until the menu is expanded. This is useful for cases when there's a lot of data, and normally no need to retrieve it, only when the user interacts with the select component.

Neither the focus event nor the click event can achieve this on their own, although it may be possible to combine them in a non-trivial way to find out when the menu is opened according to [this](https://stackoverflow.com/questions/20683587/event-for-select-box-dropdown) stack overflow discussion.

PS. I'm sure other events such as menu closing, menu closed, menu opened, etc.. would be useful as well.

**Comment by asyncliz** [comment]
@e111077 might know more about the risks, but a naive solution could be to compose and bubble those events

**Reply by e111077** [reply]
risks are that opening open close and closing are quite common event names for things that select can be in, like another menu or dialog. We'd likely have to redispatch and workshop the name such as `select-opening` or `select-open`.

**Reply by MoaLeonhardt** [reply]
Sounds like a great solution. This feature would definitely put the material select components above the regular HTML selects in developer experience.

**Reply by e111077** [reply]
Will draft up a CL / PR and see what the team thinks

**Reply by e111077** [reply]
Discussion will likely happen in internal code review but you can follow here:

https://github.com/material-components/material-web/pull/4799

**Reply by e111077** [reply]
Got a +1 from Andrew. Closing this discussion as an adopted feature! you'll see it in pre.17 release

**Reply by e111077** [reply]
update, there is a PR out to remove the selected-* prefix. I did not realize that these do not bubble anymore

**Comment by e111077** [comment]
Feature adopted in #4799

**Comment by MoaLeonhardt** [comment]
That's wonderful news! Thanks a lot! 🙂
